web/themes/custom/novel/novel.theme

```diff
@@ -96,6 +103,65 @@ function novel_preprocess_dpl_react_app__search_result(array &$variables): void
  *   The variables for the theme hook.
  */
 function novel_preprocess(array &$variables): void {
-  $baseIconPath = $variables['directory'] . '/assets/dpl-design-system/icons';
-  $variables['baseIconPath'] = $baseIconPath;
+  $directory = $variables['directory'] ?? 'web/themes/custom/novel';
+  $variables['baseIconPath'] = $directory . '/assets/dpl-design-system/icons';
+}
+
+/**
+ * Implements hook_preprocess_page().
+ *
+ * Simplify setting of logo titles and URLs using the theme settings and
+ * defaults.
+ *
+ * @param mixed[] $variables
+ *   The variables for the theme hook.
+ */
+function novel_preprocess_page(array &$variables): void {
+  $variables['logo'] = [
+    '#theme' => 'novel_logo',
+    '#title' => theme_get_setting('logo_title') ?? 'Logo title (bold)',
+    '#place' => theme_get_setting('logo_place') ?? 'Logo place',
+  ];
+
+  // Directly assign logo URL if it's set.
+  if ($logo_path = theme_get_setting('logo.path')) {
+    $variables['logo']['#image'] = [
+      '#theme' => 'image_style',
+      '#style_name' => 'logo',
+      '#uri' => $logo_path,
+    ];
+  }
+}
+
+/**
+ * Implements hook_form_system_theme_settings_alter().
+ *
+ * Ensure form settings for logo are initialized and provide a UI for logo
+ * title.
+ *
+ * @param array<string, mixed> $form
+ *   An associative array containing the structure of the form.
+ * @param \Drupal\Core\Form\FormStateInterface $form_state
+ *   The current state of the form.
+ */
+function novel_form_system_theme_settings_alter(array &$form, FormStateInterface $form_state): void {
+  $form['logo']['settings'] = $form['logo']['settings'] ?? [];
+
+  $form['logo']['settings']['logo_title'] = [
+    '#type' => 'textfield',
+    '#title' => t('Title', [], ['context' => 'Logo settings']),
+    '#default_value' => theme_get_setting('logo_title') ?? 'Logo title (bold)',
+    '#description' => t("Enter the text that will be shown as the logo on smaller screens.", [], ['context' => 'Logo settings']),
+  ];
+  $form['logo']['settings']['logo_place'] = [
+    '#type' => 'textfield',
+    '#title' => t('Place', [], ['context' => 'Logo settings']),
+    '#default_value' => theme_get_setting('logo_place') ?? 'Logo place',
+    '#description' => t("Text displayed under the logo title.", [], ['context' => 'Logo settings']),
+  ];
```

Comment:
Consider changing the context to "Novel settings"

I see that you in #595 use that context name. I suggest that we also use that here to avoid context many contexts.